controllers/gitopsservice_controller.go

```diff
@@ -482,7 +521,14 @@ func (r *ReconcileGitopsService) reconcileBackend(gitopsserviceNamespacedName ty
 		if err := controllerutil.SetControllerReference(instance, deploymentObj, r.Scheme); err != nil {
 			return reconcile.Result{}, err
 		}
-
+		if instance.Spec.InfraNodeEnabled {
+			deploymentObj.Spec.Template.Spec.NodeSelector = map[string]string{
+				"node-role.kubernetes.io/infra": "",
```

Comment:
Since this label is repeating in many places, would it make sense to instead define it in `common/common.go` and reference it from there? Maybe more maintainable for the future

cc @iam-veeramalla